perf/src/server.rs

```diff
@@ -75,7 +82,21 @@ async fn run(opt: Opt) -> Result<()> {
     let mut endpoint = quinn::EndpointBuilder::default();
     endpoint.listen(server_config);
 
-    let (endpoint, mut incoming) = endpoint.bind(&opt.listen).context("binding endpoint")?;
+    let socket = Socket::new(Domain::ipv6(), Type::dgram(), Some(Protocol::udp()))
+        .context("create socket")?;
+    socket
+        .bind(&socket2::SockAddr::from(opt.listen))
+        .context("binding endpoint")?;
+    socket
+        .set_send_buffer_size(opt.send_buffer_size)
+        .context("send buffer size")?;
+    socket
+        .set_recv_buffer_size(opt.recv_buffer_size)
+        .context("receive buffer size")?;
```

Comment:
There's significant risk of confusion since most distros by default enforce a maximum much lower than the default values set here, which cannot be modified without elevated privileges.

Reply:
I checked that it won't fail even if the OS max is lower. It will just cap the value.

Manually setting the option makes the testing a bit more resilient on platforms which have a higher `_max` and low `_default` socket buffer settings, which are a couple of production platforms I encountered. So I think its reasonable trying to do it.

If the maximum setting is too low, then this obviously won't help and still impact benchmark results. Maybe there can be another piece of code which reads the buffer size back and prints a warning if too low.

Reply:
Right, my concern is that it's easy to assume the options are doing something when, in the common case, they aren't. A warning would be perfect.